source/dem/list_insertion.cc

```diff
@@ -0,0 +1,130 @@
+#include <dem/list_insertion.h>
+
+using namespace DEM;
+
+DeclException2(ListSizeCoherence,
+               int,
+               int,
+               << "Incoherent particle position lists n=" << arg1
+               << ", m=" << arg2);
+
+// The constructor of list insertion class does not accomplish anything other
+// than check if the position list are of the coherent size and to
+// create the insertion_points member
+template <int dim>
+ListInsertion<dim>::ListInsertion(
+  const DEMSolverParameters<dim> &dem_parameters,
+  const double &                  maximum_particle_diameter)
+  : remaining_particles_of_each_type(
+      dem_parameters.physical_properties.number.at(0))
+{
+  // Inializing current inserting particle type
+  current_inserting_particle_type = 0;
+
+  this->maximum_diameter = maximum_particle_diameter;
+
+  const auto &list_x = dem_parameters.insertion_info.list_x;
+  const auto &list_y = dem_parameters.insertion_info.list_y;
+  const auto &list_z = dem_parameters.insertion_info.list_z;
+
+
+  Assert(list_x.size() == list_y.size(),
+         ListSizeCoherence(list_x.size(), list_y.size()));
+  if (dim == 3)
+    {
+      Assert(list_y.size() == list_z.size(),
+             ListSizeCoherence(list_y.size(), list_z.size()));
+    }
```

Comment:
How are you sure that in 2D, he would use x and y list?

Reply:
Because in deal.II 2D simulations only have the x and y coordinate. The Z list has no meaning in 2D because points are 2D.